packages/core/react/use-chat.ts

```diff
@@ -156,13 +156,40 @@ const getStreamedResponse = async (
   }
 
   const prefixMap: PrefixMap = {}
+  const NEWLINE = '\n'.charCodeAt(0);
 
   if (isComplexMode) {
     while (true) {
-      const { done, value } = await reader.read()
-      if (done) {
+      // iterate over each line in the stream
+      let chunks: Uint8Array[] = []
+      let totalLength = 0
+      while (true) {
+        // interate over each chunk in the line
+        const { done, value: partialValue } = await reader.read()
+        if (done) {
+          break
+        }
+        chunks.push(partialValue)
+        totalLength += partialValue.length
+        if (partialValue[partialValue.length - 1] === NEWLINE) {
+          // if the last character is a newline, we have read the whole line
+          break
+        }
+      }
```

Comment:
We can definitely merge the two while loops:

```suggestion
      const { value } = await reader.read()
      if (value) {
        chunks.push(value)
        totalLength += value.length
        if (value[value.length - 1] !== NEWLINE) {
          // if the last character is not a newline, we have not read the whole JSON value
          continue
        }
      }
```